src/AutoRest.CSharp/Generation/Writers/LowLevelClientWriter.cs

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Linq;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace AutoRest.CSharp.Generation.Writers
+{
+    internal class LowLevelClientWriter
+    {
+        public void WriteClient(CodeWriter writer, Client client, BuildContext context)
+        {
+            var cs = client.Type;
+            // Client type should have public constructor with equivalent parameters not taking ClientOptions type as last argument
+            writer.Line ($"#pragma warning disable AZC0007\n");
+
+            using (writer.Namespace(cs.Namespace + ".Protocol"))
+            {
+                writer.WriteXmlDocumentationSummary(client.Description);
+                using (writer.Scope($"{client.Declaration.Accessibility} partial class {cs.Name}"))
+                {
+                    WriteClientFields(writer, client, context);
+                    WriteClientCtors(writer, client, context);
+
+                    foreach (var clientMethod in client.Methods)
+                    {
+                        WriteClientMethod(writer, clientMethod, true, false);
+                        WriteClientMethod(writer, clientMethod, true, true);
+                        WriteClientMethod(writer, clientMethod, false, false);
+                        WriteClientMethod(writer, clientMethod, false, true);
+                        WriteClientMethodRequest(writer, clientMethod);
+                    }
+
+                    writer.Line();
+                    writer.Line($"private static JsonData ToJsonData(object value)");
+                    writer.Line($"{{");
+                    writer.Line($"    if (value is JsonData)");
+                    writer.Line($"    {{");
+                    writer.Line($"        return (JsonData)value;");
+                    writer.Line($"    }}");
+                    writer.Line($"    return new JsonData(value);");
+                    writer.Line($"}}");
+                }
+            }
+        }
+
+        private void WriteClientMethodRequest(CodeWriter writer, ClientMethod clientMethod)
+        {
+            RequestClientWriter.WriteRequestCreation(writer, clientMethod.RestClientMethod, lowLevel: true);
+        }
+
+        private void WriteClientMethod(CodeWriter writer, ClientMethod clientMethod, bool async, bool useDynamic)
+        {
+            var parameters = clientMethod.RestClientMethod.Parameters;
+
+            CSharpType? bodyType = clientMethod.RestClientMethod.ReturnType;
+            string responseType = async ? "Task<DynamicResponse>" : "DynamicResponse";
```

Comment:
Prefer using CSharpType for type manipulation.

https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Generation/Writers/ModelWriter.cs#L98-L99

as an example.